src/runtime/GlobalObjectBuiltinSymbol.cpp

```diff
@@ -101,6 +101,28 @@ Value builtinSymbolKeyFor(ExecutionState& state, Value thisValue, size_t argc, V
     return Value();
 }
 
+static Value builtinSymbolDescriptionGetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Value newTarget)
+{
+    if (thisValue.isSymbol()) {
+        if (thisValue.asSymbol()->description() == String::emptyString) {
```

Comment:
We don't guarantee that emptyString is always String::emptyString
so you can use thisValue.asSymbol().length() == 0

Reply:
If we use` length() == 0` instead of `description() == String::emptyString` then few test-cases will fail. Since there is two way to construct a Symbol's description, first if `argv[0]` is not present or it is undefined then description is going to be a `String::emptyString` in this cases the output should be `undefined`, but if argv[0] is present and it's not undefined the the description value is going to be the value of `argv[0].toString`. 

For example:

`var emptyString = new Symbol('');`, then argv[0] is going to be exist and it's not undefined, but if we use `length() == 0` in `builtinSymbolDescriptionGetter` then it is going to be true and the output will be undefined which is not correct, since the `description` String value is ''  (which is zero length String) 

 But if you have a better idea or suggestion, i'm open to it.

Reply:
IMO we can store symbol description Optional<String*> instead of String*.
store nullptr in Optional can means undefoned or null.
how about this one?